components/micro-gateway-core/src/main/ballerina/src/gateway/handlers/token_introspection_handler.bal

```diff
@@ -91,7 +91,7 @@ public type KeyValidationHandler object {
                 any clientId = claims[CLIENT_ID];
                 boolean isAllowed = false;
                 // If validateSubscription is true and clientID is present, do the subscription validation.
-                if (clientId != () && clientId is string && self.validateSubscriptions) {
+                if (clientId != () && clientId is string) {
```

Comment:
When subscription validation is enabled and the client id is missing, we need to handle that error scenario as well.

Reply:
https://github.com/wso2/product-microgateway/pull/1331/commits/94b285498042de08d4df06fe949a5b1e18af1c78